ducktape/command_line/parse_args.py

```diff
@@ -0,0 +1,115 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.command_line.defaults import ConsoleDefaults
+from ducktape.utils.util import ducktape_version
+
+import argparse
+import itertools
+import os
+import sys
+
+
+def create_ducktape_parser():
+    parser = argparse.ArgumentParser(description="Discover and run your tests")
+    parser.add_argument('test_path', metavar='test_path', type=str, nargs='*', default=[os.getcwd()],
+                        help='one or more space-delimited strings indicating where to search for tests.')
+    parser.add_argument("--collect-only", action="store_true", help="display collected tests, but do not run.")
+    parser.add_argument("--debug", action="store_true", help="pipe more verbose test output to stdout.")
+    parser.add_argument("--config-file", action="store", default=ConsoleDefaults.USER_CONFIG_FILE,
+                        help="path to project-specific configuration file.")
+    parser.add_argument("--cluster", action="store", default=ConsoleDefaults.CLUSTER_TYPE,
+                        help="cluster class to use to allocate nodes for tests.")
+    parser.add_argument("--cluster-file", action="store", default=None,
+                        help="path to a json file which provides information needed to initialize a json cluster. "
+                             "The file is used to read/write cached cluster info if "
+                             "cluster is ducktape.cluster.vagrant.VagrantCluster.")
+    parser.add_argument("--results-root", action="store", default=ConsoleDefaults.RESULTS_ROOT_DIRECTORY,
+                        help="path to custom root results directory. Running ducktape with this root "
+                             "specified will result in new test results being stored in a subdirectory of "
+                             "this root directory.")
+    parser.add_argument("--exit-first", action="store_true", help="exit after first failure")
+    parser.add_argument("--no-teardown", action="store_true",
+                        help="don't kill running processes or remove log files when a test has finished running. "
+                             "This is primarily useful for test developers who want to interact with running "
+                             "services after a test has run.")
+    parser.add_argument("--version", action="store_true", help="display version")
+    parser.add_argument("--parameters", action="store",
+                        help="inject these arguments into the specified test(s). Specify parameters as a JSON string.")
+    parser.add_argument("--globals", action="store",
```

Comment:
I think so, but to clarify: how does a project make use of the globals exactly? Is the expectation that we define defaults for the globals directly in Python code and the ones passed in here (or via config file) are used to override those, e.g. something like

```
my_var = globals.get("my-config", DEFAULT_MY_CONFIG)
```

And then only if you were working in a non-default environment would you need to pass in something for the `--globals` command line config? Basically, I'm trying to make sure `--globals` in the use case you're envisioning is only used in special cases and that, e.g., if I were just running the kafka tests from a standard checkout, I wouldn't need to add a `--globals` flag.